app/controllers/spree/api/v2/storefront/product_questions_controller.rb

```diff
@@ -17,7 +17,7 @@ def show
           def create
             @product_question = Spree::ProductQuestion.new(product_question_params)
             @product_question.product = @product
-            @product_question.user = spree_current_user if spree_user_signed_in?
+            @product_question.user = spree_current_user
```

Comment:
you did not check if anonymous question is allowed or not while creating question?

Reply:
It is checked in before_action